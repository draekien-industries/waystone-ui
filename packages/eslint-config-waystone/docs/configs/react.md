# eslint-config-waystone - react

## Usage

```js
// .eslintrc.js
module.exports = {
  extends: ['waystone/react'],
};
```

## Env

- browser
- es6
- shared-node-browser

## Extends

- airbnb
- turbo
- eslint:recommended
- import/recommended
- import/react
- import/typescript
- @typescript-eslint/eslint-recommended
- @typescript-eslint/recommended
- react/recommended
- react/jsx-runtime
- waystone/recommended
- jsx-a11y/recomended
- testing-library/recommended
- jest/recommended
- prettier
