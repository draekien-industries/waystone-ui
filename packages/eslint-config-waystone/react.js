module.exports = {
  env: {
    browser: true,
    es6: true,
    'shared-node-browser': true,
  },
  extends: [
    'airbnb',
    'turbo',
    'eslint:recommended',
    'plugin:import/recommended',
    'plugin:import/react',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/eslint-recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:react/recommended',
    'plugin:react/jsx-runtime',
    'plugin:react-hooks/recommended',
    'plugin:waystone/recommended',
    'plugin:jsx-a11y/recommended',
    'prettier',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
    ecmaVersion: 'latest',
    sourceType: 'module',
  },
  settings: {
    react: {
      version: 'detect',
    },
  },
  rules: {
    'no-plusplus': ['error', { allowForLoopAfterthoughts: true }],
    '@typescript-eslint/consistent-type-definitions': ['error', 'type'],
    '@typescript-eslint/consistent-type-imports': [
      'error',
      {
        prefer: 'type-imports',
        fixStyle: 'separate-type-imports',
      },
    ],
    '@typescript-eslint/no-unused-vars': [
      'error',
      { varsIgnorePattern: '^_*' },
    ],
    'react/function-component-definition': [
      'warn',
      {
        namedComponents: 'arrow-function',
        unnamedComponents: 'arrow-function',
      },
    ],
    'react/jsx-filename-extension': ['warn', { extensions: ['.tsx'] }],
    'react/jsx-props-no-spreading': 'off',
    'react/no-unknown-property': ['error', { ignore: ['sx'] }],
    'react/require-default-props': 'off',
    'import/extensions': [
      'error',
      'ignorePackages',
      { ts: 'never', tsx: 'never', js: 'never', jsx: 'never' },
    ],
    'import/prefer-default-export': 'off',
    'import/no-extraneous-dependencies': ['error', { devDependencies: true }],
    'no-use-before-define': [
      'error',
      {
        functions: false,
        classes: true,
        variables: true,
        allowNamedExports: false,
      },
    ],
  },
  overrides: [
    {
      env: { jest: true },
      files: [
        '**/__tests__/**/*.[jt]s?(x)',
        '**/__fixtures__/**/*.[jt]s?(x)',
        '**/?(*.)+(spec|test).[jt]s?(x)',
      ],
      extends: [
        'airbnb',
        'turbo',
        'eslint:recommended',
        'plugin:import/recommended',
        'plugin:import/typescript',
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:react/recommended',
        'plugin:react/jsx-runtime',
        'plugin:react-hooks/recommended',
        'plugin:waystone/recommended',
        'plugin:jsx-a11y/recommended',
        'plugin:testing-library/react',
        'plugin:jest/recommended',
        'prettier',
      ],
      rules: {
        'import/no-extraneous-dependencies': [
          'off',
          { devDependencies: ['**/?(*.)+(spec|test).[jt]s?(x)'] },
        ],
        'react/jsx-filename-extension': ['warn', { extensions: ['.tsx'] }],
        'import/extensions': [
          'error',
          'ignorePackages',
          { ts: 'never', tsx: 'never', js: 'never', jsx: 'never' },
        ],
      },
    },
  ],
  ignorePatterns: [
    '**/*.js',
    '**/*.json',
    '**/*.d.ts',
    '**/*.g.*',
    '**/*.generated.*',
    'node_modules',
    'public',
    'styles',
    '.next',
    'coverage',
    'dist',
    '.turbo',
  ],
};
