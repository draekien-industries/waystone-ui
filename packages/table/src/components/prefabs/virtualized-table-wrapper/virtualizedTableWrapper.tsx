import type { DimensionsAttributes } from '@waystone/types';
import {
  forwardRef,
  type ForwardRefExoticComponent,
  type ForwardedRef,
  type Key,
  type ReactNode,
  type RefAttributes,
} from 'react';

export type VirtualizedTableWrapperProps = {
  children: ReactNode;
} & DimensionsAttributes;

const VirtualizedTableWrapperWithRef = (
  { children, height = 800, ...rest }: VirtualizedTableWrapperProps,
  ref: ForwardedRef<HTMLDivElement>
) => (
  <div
    ref={ref}
    sx={{
      overflow: 'auto',
      position: 'relative',
      height,
      ...rest,
    }}
  >
    {children}
  </div>
);

export const VirtualizedTableWrapper: ForwardRefExoticComponent<
  VirtualizedTableWrapperProps & {
    key?: Key | null | undefined;
    ref: RefAttributes<HTMLDivElement>['ref'];
  }
> = forwardRef(VirtualizedTableWrapperWithRef);
