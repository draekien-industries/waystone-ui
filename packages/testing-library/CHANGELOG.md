# @waystone/testing-library

## 9.2.0

### Minor Changes

- 21db598: chore: bump deps and add documentation

### Patch Changes

- Updated dependencies [21db598]
  - @waystone/theme@9.3.0

## 9.1.6

### Patch Changes

- 41664e2: fix: use type imports to improve bundling
- Updated dependencies [41664e2]
  - @waystone/theme@9.2.5

## 9.1.5

### Patch Changes

- 856ba2b: fix: missing dependencies when installing package
- Updated dependencies [856ba2b]
  - @waystone/theme@9.2.4

## 9.1.4

### Patch Changes

- 5b726aa: fix build and dependencies
- Updated dependencies [5b726aa]
  - @waystone/theme@9.2.3

## 9.1.3

### Patch Changes

- 7086019: Update dependencies
- Updated dependencies [7086019]
  - @waystone/theme@9.2.2

## 9.1.2

### Patch Changes

- @waystone/theme@9.2.1

## 9.1.1

### Patch Changes

- Updated dependencies [8f75dd6]
  - @waystone/theme@9.2.0

## 9.1.0

### Minor Changes

- 5c43357: Switch to pnpm and implement minimal syntax highlighter

### Patch Changes

- Updated dependencies [5c43357]
  - @waystone/theme@9.1.0

## 9.0.0

### Major Changes

- 4a19c42: Update package structure

  Created the `@waystone/ui` package for easier installation and created the `@waystone/testing-library`
  package for a preconfigured testing-library render function.

  Uninstall your `@waystone/*` packages, and run the command `yarn add @waystone/ui`
  to update to the latest version. You will need to update the imports to point
  to the new source.

### Patch Changes

- Updated dependencies [4a19c42]
  - @waystone/theme@9.0.0
