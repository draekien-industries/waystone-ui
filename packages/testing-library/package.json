{
  "name": "@waystone/testing-library",
  "version": "10.0.2",
  "private": true,
  "main": "./src/index.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/draekien-industries/waystone-ui"
  },
  "description": "The react component library used by draekien-industries",
  "author": "draekien-industries",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "src/**"
  ],
  "scripts": {
    "lint": "eslint --ext .tsx,.ts,.jsx,.js ./src/ --fix",
    "lint:release": "eslint --ext .tsx,.ts,.jsx,.js ./src/"
  },
  "devDependencies": {
    "@waystone/tsconfig": "workspace:*",
    "eslint-config-waystone": "workspace:*",
    "eslint-plugin-waystone": "workspace:*",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "dependencies": {
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.2.0",
    "@testing-library/user-event": "^14.6.0",
    "@theme-ui/color": "^0.17.1",
    "@theme-ui/css": "^0.17.1",
    "@waystone/theme-provider": "workspace:*",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "theme-ui": "^0.17.1"
  },
  "publishConfig": {
    "access": "public"
  }
}
