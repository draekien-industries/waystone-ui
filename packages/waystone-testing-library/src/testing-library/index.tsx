'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
