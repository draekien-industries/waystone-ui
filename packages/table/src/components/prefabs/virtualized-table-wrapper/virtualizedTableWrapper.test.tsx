import { render, renderHook } from '@waystone/testing-library';
import { useRef } from 'react';
import { VirtualizedTableWrapper } from './virtualizedTableWrapper';

describe('<VirtualizedTableWrapper />', () => {
  it('requires ref', () => {
    const { container } = render(
      // @ts-expect-error ref is required
      <VirtualizedTableWrapper>test</VirtualizedTableWrapper>
    );

    expect(container).toMatchSnapshot();
  });

  it('accepts ref', () => {
    const { result } = renderHook(() => useRef<HTMLDivElement>(null));
    const { container } = render(
      <VirtualizedTableWrapper ref={result.current}>
        test
      </VirtualizedTableWrapper>
    );

    expect(container).toMatchSnapshot();
  });

  it('accepts height', () => {
    const { result } = renderHook(() => useRef<HTMLDivElement>(null));
    const { container } = render(
      <VirtualizedTableWrapper
        ref={result.current}
        height={1000}
        minHeight={500}
        maxHeight={1500}
      >
        test
      </VirtualizedTableWrapper>
    );

    expect(container).toMatchSnapshot();
  });

  it('accepts width', () => {
    const { result } = renderHook(() => useRef<HTMLDivElement>(null));
    const { container } = render(
      <VirtualizedTableWrapper
        ref={result.current}
        width={1000}
        minWidth={500}
        maxWidth={1500}
      >
        test
      </VirtualizedTableWrapper>
    );

    expect(container).toMatchSnapshot();
  });
});
