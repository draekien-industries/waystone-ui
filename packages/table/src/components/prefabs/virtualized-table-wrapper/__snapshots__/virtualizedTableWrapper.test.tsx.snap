// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VirtualizedTableWrapper /> accepts height 1`] = `
<div>
  <div
    class="css-1pgk7dz"
  >
    test
  </div>
</div>
`;

exports[`<VirtualizedTableWrapper /> accepts ref 1`] = `
<div>
  <div
    class="css-eagjw6"
  >
    test
  </div>
</div>
`;

exports[`<VirtualizedTableWrapper /> accepts width 1`] = `
<div>
  <div
    class="css-wmn3kp"
  >
    test
  </div>
</div>
`;

exports[`<VirtualizedTableWrapper /> requires ref 1`] = `
<div>
  <div
    class="css-eagjw6"
  >
    test
  </div>
</div>
`;
